app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarMenu.kt

```diff
@@ -22,7 +22,7 @@ interface ToolbarMenu {
         object AddToTopSites : Item()
         object InstallPwaToHomeScreen : Item()
         object AddToHomeScreen : Item()
-        data class SyncAccount(val signedIn: Boolean) : Item()
+        data class SyncAccount(val existingAccount: Boolean, val signedIn: Boolean) : Item()
```

Comment:
I think we should go with the same language `isAuthenticatedAccount` since `existingAccount` changes the meaning of it from the original intent. I prefix it with `is` to make it super obvious that it represents a `Boolean` when reading the code.

Reply:
I think rather the naming used in fxa should be updated to more closely express the current status of the account.
The documentation for `authenticatedAccount` [says](https://github.com/mozilla-mobile/android-components/blob/bfaf514793dd20f9f12d3f551f9ff3e470f52727/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/FxaAccountManager.kt#L325-L338)
"..Returned [OAuthAccount] may need to be re-authenticated."

Based on this and my below comment about the logic used in the app and tests `existingAccount` offers more info about the current status while `authenitcatedAccount` but not signed in seems confusing.

Reply:
Maybe `accountExists` expresses better that that parameter is for a Boolean?

Reply:
For this instance, `authenticatedAccount` is defined in https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/components/accounts/FenixAccountManager.kt#19. It is slightly odd that we used the same naming as the account manager but don't technically return the `OAuthAccount`.

You're right that the authenticated account might need re-auth, but I am not too particular about that detail since it is in some sort of authenticated state. My main point would be to persist the same naming of instead of changing it midway in the ideal case. That said the naming is open to change - I don't particularly see a problem with simply persisting with the `authenticatedAccount` naming due to the fact that it is still in sort of authenticated state and it is known that consumers should check if it needs to re-auth. 

So, I think it boils down to the use case here, which we basically have tri-state (Authenticated, ReAuth, Not Signed in). Perhaps, we should change `FenixAccountManager` to return something that is super clear about the state instead of us having to basically do the same thing all the time with getting the `authenticatedAccount` and checking `signedIn`. If you look at `FenixAccountManager`, the whole logic is kinda duplicated multiple times. 

Feel free to ping me if this wasn't clear. I was going through the code, thinking, writing and revising as my thinking was kinda all over the place.

Reply:
Maybe `authenticatedAccountExists` would be a more appropriate parameter name?

Reply:
After talking with Gabriel I've added a new `accountState` property to FenixAccountManager and will use this simpler way of inferring the directions to which to navigate to.